src/main/java/com/criticalsoftware/AnnouncementRequest.java

```diff
@@ -0,0 +1,46 @@
+package com.criticalsoftware;
+
+import jakarta.validation.Valid;
+import jakarta.validation.constraints.NotBlank;
+import jakarta.validation.constraints.NotNull;
+import jakarta.json.bind.annotation.JsonbDateFormat;
+import java.time.LocalDateTime;
+
+public class AnnouncementRequest {
+
+    @NotBlank(message = "User donor ID is mandatory")
+    private String userDonorId;
+
+    @NotNull(message = "Date is mandatory")
+    @JsonbDateFormat("yyyy-MM-dd'T'HH:mm:ss'Z'")
+    private LocalDateTime date;
+
+    @Valid
+    @NotNull(message = "Produto is mandatory" )
+    private Product product;
```

Comment:
Product should not be here. Add the fileds in Product here, not the whole object.

Reply:
Done!

Reply:
Done!